tensorflow_probability/python/positive_semidefinite_kernels/rational_quadratic.py

```diff
@@ -129,24 +130,27 @@ def __init__(
             tf.assert_positive(
                 self._scale_mixture_rate)] if validate_args else []):
           self._scale_mixture_rate = tf.identity(self._scale_mixture_rate)
+      self._scale_mixture_rate = _validate_arg_if_not_none(scale_mixture_rate, tf.assert_positive, validate_args)
```

Comment:
This should either take self._scale_mixture_rate (since the positivity check is embedded above in to that), or all the statements above should only modify scale_mixture_rate.

Reply:
The same below for every other block

Reply:
Sure, just have fixed it

Reply:
Sorry, can you remove all the control_dependencies lines? _validate_arg_if_not_none already does that. The minimal block should be:

```suggestion
if amplitude is not None:
  amplitude = tf.convert_to_tensor(amplitude, name="amplitude", dtype=dtype)
self._amplitude = _validate_args_if_not_none(
  amplitude, tf.assert_positive, validate_args)
```

Reply:
sure